haskell/experimental/private/module.bzl

```diff
@@ -200,8 +214,10 @@ def _build_haskell_module(
             package_ids = hs.package_ids,
             package_databases = depset(
                 transitive = [
-                    dep_info.package_databases,
+                    # Mind the order in which databases are specified here.
+                    # See Note [Deps as both narrowed and not narrowed].
                     narrowed_deps_info.empty_lib_package_databases,
+                    dep_info.package_databases,
                 ],
```

Comment:
hm, in some tests this is still not producing the environment file with the flags in the order given here.
Can I set `order = "postorder"` here, regardless of the order of the transitive depsets?

Reply:
I had to use preorder here and at another place. It seem to work now.

Reply:
The [documentation](https://docs.bazel.build/versions/main/skylark/lib/depset.html) says the following about the `order` parameter:

> Two depsets may only be merged if either both depsets have the same order, or one of them has "default" order. In the latter case the resulting depset's order will be the same as the other's order.